x/concentrated-liquidity/position.go

```diff
@@ -226,3 +228,156 @@ func (k Keeper) getNextPositionIdAndIncrement(ctx sdk.Context) uint64 {
+	if err != nil {
+		return 0, err
+	}
+
+	fullyChargedDuration := types.SupportedUptimes[len(types.SupportedUptimes)-1]
+
+	// The new position's timestamp is the current block time minus the fully charged duration.
+	joinTime := ctx.BlockTime().Add(-fullyChargedDuration)
+
+	// Get the next position ID and increment the global counter.
+	newPositionId := k.getNextPositionIdAndIncrement(ctx)
+
+	// Initialize the fee accumulator for the new position.
+	if err := k.initializeFeeAccumulatorPosition(ctx, poolId, lowerTick, upperTick, newPositionId); err != nil {
+		return 0, err
+	}
+
+	// Check if the position already exists.
+	hasFullPosition := k.hasFullPosition(ctx, newPositionId)
+	if !hasFullPosition {
+		// If the position does not exist, initialize it with the provided liquidity and tick range.
+		err = k.initOrUpdatePositionUptime(ctx, poolId, liquidity, owner, lowerTick, upperTick, sdk.ZeroDec(), joinTime, newPositionId)
+		if err != nil {
+			return 0, err
+		}
+	} else {
+		// If the position already exists, return an error.
+		return 0, err
+	}
+
+	// Update the position in the pool based on the provided tick range and liquidity delta.
+	_, _, err = k.updatePosition(ctx, poolId, owner, lowerTick, upperTick, liquidity, joinTime, newPositionId)
+	if err != nil {
+		return 0, err
+	}
+
+	// Get the new position
+	newPosition, err := k.GetPosition(ctx, newPositionId)
+	if err != nil {
+		return 0, err
+	}
+
+	// Get the new position's store name as well as uptime accumulators for the pool.
+	newPositionName := string(types.KeyPositionId(newPositionId))
+	uptimeAccumulators, err := k.getUptimeAccumulators(ctx, newPosition.PoolId)
+	if err != nil {
+		return 0, err
+	}
+
+	// Move unclaimed rewards from the old positions to the new position.
+	// Also, delete the old positions from state.
+
+	// Compute uptime growth outside of the range between lower tick and upper tick
+	uptimeGrowthOutside, err := k.GetUptimeGrowthOutsideRange(ctx, newPosition.PoolId, newPosition.LowerTick, newPosition.UpperTick)
+	if err != nil {
+		return 0, err
+	}
+
+	// Move unclaimed rewards from the old positions to the new position.
+	// Also, delete the old positions from state.
+
+	// Loop through each position ID.
+	for _, positionId := range positionIds {
+		// Loop through each uptime accumulator for the pool.
+		for uptimeIndex, uptimeAccum := range uptimeAccumulators {
+			oldPositionName := string(types.KeyPositionId(positionId))
+			// Check if the accumulator contains the position.
+			hasPosition, err := uptimeAccum.HasPosition(oldPositionName)
+			if err != nil {
+				return 0, err
+			}
+			// If the accumulator contains the position, move the unclaimed rewards to the new position.
+			if hasPosition {
+				// Prepare the accumulator for the old position.
+				rewards, dust, err := prepareAccumAndClaimRewards(uptimeAccum, oldPositionName, uptimeGrowthOutside[uptimeIndex])
```

Comment:
We should manually delete the accumulators upon claiming because `accum/ClaimRewards` only deletes the accumulators if we have zero num shares.

We cannot manually set the num shares to zero because that would break the claiming logic

Reply:
Added manual delete position here. Also added tests to check that accum position AND cl positions are deleted post fungify call https://github.com/osmosis-labs/osmosis/pull/4800/commits/93aa0f949cfd7c93f86a5901fb5ea8c387ec8d6d